kernel/include/syscall_handler.h

```diff
@@ -287,11 +273,15 @@ bool z_syscall_verify_msg(bool expr, const char *fmt, ...)
  *             oops
  * @param fmt Printf-style format string (followed by appropriate variadic
  *            arguments) to print on verification failure
- * @return 0 on success, nonzero on failure
+ * @return False on success, True on failure
  */
-#define Z_SYSCALL_VERIFY_MSG(expr, fmt, ...) \
-	z_syscall_verify_msg(!(expr), "syscall %s failed check: " fmt "\n", \
-			     __func__, ##__VA_ARGS__)
+#define Z_SYSCALL_VERIFY_MSG(expr, fmt, ...) ({ \
+	bool expr_copy = !(expr); \
+	if (expr_copy) { \
+		printk("syscall %s failed check: " fmt "\n", \
+		       __func__, ##__VA_ARGS__); \
+	} \
+	expr_copy; })
```

Comment:
The problem with this solution is that we may trigger other problem when you call it with the variadic parameter is empty. This violate C99 standard (and C11 AFAIK too).

```
warning: ISO C99 requires at least one argument for the "..." in a variadic macro
```

Honestly I don't know which one is worse for us.

Reply:
what?

printf("foo!") is not valid?

at any rate we pass __func__ to printk so what's the problem?

Reply:
what is not valid is:
```
Z_SYSCALL_VERIFY_MSG(boolean_x, "invalid");
```
though it's valid:
```
Z_SYSCALL_VERIFY_MSG(boolean_x, "invalid number: %d", y);
```

Reply:
Oh, I see what you mean now.

This does not produce a warning because I am using a GNU extension (special meaning of ##) https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html

IIRC, not using stdard is mandatory, and not using GCC compiler extensions is advisory, so it seems pretty clear which one is worse